src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultBindingMetadataProvider.cs

```diff
@@ -143,7 +143,8 @@ private static ConstructorInfo GetRecordTypeConstructor(Type type, ConstructorIn
             static bool IsRecordType(Type type)
             {
                 // Based on the state of the art as described in https://github.com/dotnet/roslyn/issues/45777
-                var cloneMethod = type.GetMethod("<>Clone", BindingFlags.Public | BindingFlags.Instance);
+                var cloneMethod = type.GetMethod("<Clone>$", BindingFlags.Public | BindingFlags.Instance) ??
+                    type.GetMethod("<>Clone", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Isn't this fixing an extreme corner case❔ I wouldn't expect users to build with an RC1 compiler and preview8 MVC.

Reply:
There's a chance that the compiler in VS is newer and causes this to break. This would be a defensive change for those scenarios.